packages/jaeger-ui/src/model/link-patterns.tsx

```diff
@@ -139,6 +139,35 @@ function callTemplate(template: ProcessedTemplate, data: any) {
   return template.template(data);
 }
 
+export function computeTraceLinks(linkPatterns: ProcessedLinkPattern[], trace: Trace) {
+  const result: { url: string; text: string }[] = [];
+  const validKeys = Object.keys(trace)
+    // @ts-ignore
+    .filter(key => typeof trace[key] === 'string' || typeof trace[key] === 'number');
+
+  linkPatterns
+    .filter(pattern => pattern.type('traces'))
+    .forEach(pattern => {
+      const parameterValues: Record<string, any> = {};
+      const allParameters = pattern.parameters.every(parameter => {
+        if (validKeys.some(name => name === parameter)) {
+          // @ts-ignore
```

Comment:
this `@ts-ignore` will require the following line to be:
`parameterValues[parameter] = trace[(parameter as keyof Trace)];`

`as` is safer than `@ts-ignore` and unfortunately necessary here. In this place, you should also add a comment saying we know parameter is a `keyof Trace` because it is in `validKeys`